### PR TITLE
Add muc light and token reconnection to docs

### PR DESCRIPTION
**Pull request by michalwski**
This PR includes:
* added token-based reconnection to docs
* added MUC light to docs

### CODE REVIEW

**Comment by michalwski** [comment]
Rendered version of token reconnection is here: http://mongooseim.readthedocs.io/en/custom-extenstions-in-doc/user-guide/token-reconnection/

**Comment by michalwski** [comment]
And here goes MUC light spec: http://mongooseim.readthedocs.io/en/custom-extenstions-in-doc/user-guide/muc_light/

**Comment by michalwski** [comment]
I have a question regarding the section in which `MUC light` and `Token auth` should be places. Currently I added them to `User Guide` but maybe there is a better (possible new place). I'd like to here your opinion.

File: doc/user-guide/muc_light.md [hunk 1]
```diff
@@ -0,0 +1,1821 @@
...
+1. A user may block the attempts of being added to the specific room or by specific user.
+1. The message sent in the room is always broadcasted to every occupant.
+1. The full occupant list is always available to all occupants.
+1. Occupant is always visible on the list, even if it does not have any resources online.
+1. Occupants can only have two affiliations: owner and member.
+1. There must be at most one owner in the room (the service can choose to treat all users equally).
+1. If the room becomes empty, it is destroyed.
+1. Occupants cannot hide behind nicks. Their real bare JID is always visible to everyone
+1. No exchange of any <presence/> stanza inside the room.
+1. The user MUST be able to retrieve the list of rooms he or she occupies.
```
Comment (ludwikbukowski) [inline doc/user-guide/muc_light.md]: uppercase "MUST" is not necessary here

**Comment by Nyco** [comment]
Why not create a new sections? Like "XEPs"?

File: doc/user-guide/muc_light.md (hunk 1 above)
Comment (michalwski) [inline doc/user-guide/muc_light.md]: Why do you think so? It's taken from the original protoXEP created by @fenek. Words like `MUST`, `SHOULD`, `MAY` are usually uppercased in documents describing a standard / protocol.

Comment (Nyco) [inline doc/user-guide/muc_light.md]: "MUST" is a reserved, all uppercase keyword, right?

**Comment by michalwski** [comment]
> Why not create a new sections? Like "XEPs"?

"XEPs" is not good name for me as it indicates XEPs accepted by XSF.

File: doc/user-guide/muc_light.md (hunk 1 above)
Comment (ludwikbukowski) [inline doc/user-guide/muc_light.md]: ok, so here
https://github.com/esl/MongooseIM/pull/877#discussion-diff-69281409R18
the word "must" also should be uppercased?

Comment (michalwski) [inline doc/user-guide/muc_light.md]: Yes you are :)

**Comment by Nyco** [comment]
Like https://developers.google.com/talk/jep_extensions/extensions could be:
"MongooseIM open XMPP extensions"

File: doc/user-guide/muc_light.md [hunk 2]
```diff
@@ -0,0 +1,1821 @@
...
+
+## 6.2 XEP-0313 Message Archive Management
+
+This section defines the rules for archiving MUC Light events and messages. Stanzas described in the subsections below MUST be archived by the server. The stanzas not included here MUST NOT be archived.
+
+The <message/> element inside <forwarded> MUST include "from" attribute and MUST NOT include "to" attribute. "id" SHOULD be archived as well.
+
+In case of regular groupchat messages, "from" attribute MUST consist of a room full JID with sender bare JID in the resource part. As for room notification, e.g. create event, "from" MUST be equal to room bare JID.
+
+** protocol. Archive can be fetched only from a specific room, the client MUST NOT query MUC Light service directly.**
```
Comment (ludwikbukowski) [inline doc/user-guide/muc_light.md]: "protocol." ? :D

File: doc/user-guide/muc_light.md [hunk 3]
```diff
@@ -0,0 +1,1821 @@
...
+
+### 6.2.1 Groupchat message from occupant
+
+Message from a user MUST be archived with all child elements.
+
+**Occupant queries MAM and receives regular groupchat message**
```
Comment (ludwikbukowski) [inline doc/user-guide/muc_light.md]: Due to this issue https://github.com/esl/MongooseIM/issues/867 and as far it's not solved, are we sure we want add this section to the official doc?

**Comment by michalwski** [comment]
All right, I added the `MongooseIM open XMPP extensions`. How does it look now (on readthedocs)?

File: doc/user-guide/muc_light.md (hunk 3 above)
Comment (michalwski) [inline doc/user-guide/muc_light.md]: I'd leave it here and work on fixing the implementation. I think this is a must-have feature.

**Pull request merged**